modules/bastion/main.tf

```diff
@@ -0,0 +1,49 @@
+resource "aws_instance" "bastion" {
+  ami           = var.ami_id
+  instance_type = var.instance_type
+  key_name      = var.bastion_key_pair_name
+
+  network_interface {
+    device_index         = 0
+    network_interface_id = aws_network_interface.bastion_nic.id
+  }
+
+  root_block_device {
+    volume_size = var.os_disk_size
+  }
+
+  tags = merge(var.tags, { Name : var.bastion_instance_name })
+}
+
+resource "aws_security_group" "allow_ssh" {
```

Comment:
we want 2 security group rules where each rule is bound to a diff interface.

mgmt rules should: allow ssh
mon rules: should allow GENEVE from the VPC CIDR